table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+          break;
+        case IGNORE:
+          mIgnoredTables.add(item);
+          break;
+        case NONE:
+        default:
+          // unreachable
+      }
+      return this;
+    }
+
+    /**
+     * Include or exclude the partitions of a bypassed tabls.
+     * @param tableName name of the parent table
+     * @param mode {@link Mode} of the tables
+     * @return this builder
+     */
+    public Builder setBypassedPartitionsMode(String tableName, Mode mode) {
+      mPartitionBypassModes.put(tableName, mode);
+      return this;
+    }
+
+    /**
+     * Add a partition by its exact name.
+     * @param tableName the containing table
+     * @param partitionName partition name
+     * @return this builder
+     */
+    public Builder addBypassedPartition(String tableName, String partitionName) {
+      return addBypassedPartition(tableName, new ExactNameTemplate(partitionName));
+    }
+
+    /**
+     * Add a partition by pattern.
+     * @param tableName the containing table
+     * @param partitionPattern partition pattern
+     * @return this builder
+     */
+    public Builder addBypassedPartition(String tableName, Pattern partitionPattern) {
+      return addBypassedPartition(tableName, new PatternNameTemplate(partitionPattern));
+    }
+
+    private Builder addBypassedPartition(String tableName, NameTemplate partition) {
+      mBypassedPartitions.compute(tableName, (tblName, set) -> {
+        if (set == null) {
+          return Sets.newHashSet(partition);
+        } else {
+          set.add(partition);
+          return set;
+        }
+      });
+      return this;
+    }
+
+    /**
+     * Build a {@link UdbFilterSpec}.
+     * The builder should not be used after this method returns.
+     * @return {@link UdbFilterSpec}
+     * @throws IllegalArgumentException when invalid or conflicting configurations
+     *                                          have been provided
+     */
+    public UdbFilterSpec build() {
+      ImmutableMap.Builder<String, Pair<Mode, Set<NameTemplate>>> partitionsBuilder
+          = ImmutableMap.builder();
+      for (Map.Entry<String, Set<NameTemplate>> entry : mBypassedPartitions.entrySet()) {
+        String tableName = entry.getKey();
+        Set<NameTemplate> partitionNameTemplates = entry.getValue();
+        partitionsBuilder.put(tableName, new Pair<>(
+            mPartitionBypassModes.getOrDefault(tableName, Mode.NONE), partitionNameTemplates));
+      }
+      UdbFilterSpec built = new UdbFilterSpec(
+          mBypassedTables,
+          mTableBypassMode,
+          partitionsBuilder.build(),
+          mIgnoredTables,
+          mIgnoreMode
+      );
+      // release references to prevent reuse of builder
+      mBypassedTables = ImmutableSet.of();
+      mTableBypassMode = Mode.NONE;
+      mBypassedPartitions = ImmutableMap.of();
+      mPartitionBypassModes = ImmutableMap.of();
+      mIgnoredTables = ImmutableSet.of();
+      mIgnoreMode = Mode.NONE;
```

Comment:
I view this unnecessary. We have a lot of builder pattern in Alluxio repo. Typically we don't clean the references for builder. Following the convention is fine.

Reply:
Removed in 40f91a5798b714c552f0fc2659a3549429ca3dfa